static/js/views/discovered-addon.js

```diff
@@ -123,6 +129,60 @@ class DiscoveredAddon {
         controlDiv.innerHTML = el;
       });
   }
+  
+  /**
+   * Handle a click on the license button.
+   */
+  handleLicense(e) {
+    if(e.target.getAttribute('data-id')) {
+      
+      API.getAddonsInfo().then((data) => {
+        console.log("API.getAddonsInfo() = ");
+        console.log(data);  
+
+        const licenseUrl = data['urls'][0] + "/license/" + e.target.getAttribute('data-id');
+
+        let modal = document.getElementById('media-modal');
+
+        if(modal == null){
+          let modalContainer = document.createElement('div');
+          modalContainer.className = "media-modal";
+          modalContainer.id = "media-modal";
+
+          let modalFrame = document.createElement('div');
+          modalFrame.className = "media-modal-frame";
+          modalFrame.innerHTML = '<div class="media-modal-close" id="modal-close-button"></div><div class="media-modal-content"><p id="media-modal-text"></p></div>';
+
+          modalContainer.appendChild(modalFrame);
+          document.body.appendChild(modalContainer);
+
+          let modalCloseButton = document.getElementById('modal-close-button');
+          modalCloseButton.addEventListener(
+            'click',
+            () => {
+              modal = document.getElementById("media-modal");
+              modal.parentNode.removeChild(modal);
+            }
+          );
+
+          fetch(licenseUrl).then(response => {
+            if (!response.ok) {
+              throw new Error("HTTP error " + response.status);
+              document.getElementById("media-modal-text").innerText = fluent.getMessage('failed-read-file');
```

Comment:
That's dead code

Reply:
Could you be more specific?

Reply:
I've removed the check for an ok response.